app/models/assignment.rb

```diff
@@ -435,7 +435,7 @@ def clone_groupings_from(assessment_id)
             unless grouping.memberships << membership # this saves the membership as a side effect, i.e. can return false
               grouping.memberships.delete(membership)
               warnings << I18n.t('groups.clone_warning.no_member',
-                                 member: m.user.user_name, group: g.group.group_name, error: membership.errors.messages)
+                                 member: m.role.user.user_name, group: g.group.group_name, error: membership.errors.messages)
```

Comment:
If you add this to the Role model:

```
delegate_missing_to :user
```

Then you can access the user attributes directly from the role as `m.role.user_name` (for example). This might simplify some of your queries if you want. 

More info here:

https://apidock.com/rails/v6.1.3.1/Module/delegate_missing_to